pkg/database/handler_test.go

```diff
@@ -20,41 +21,127 @@ import (
 	"gorm.io/gorm"
 )
 
-func TestHandler_List(t *testing.T) {
-	name := "name"
+func TestHandler_FindById(t *testing.T) {
+	repository := &mockRepository{}
+	database := &model.Database{
+		GroupName: "name",
+	}
+	repository.
+		On("FindById", uint(1)).
+		Return(database, nil)
+	service := NewService(config.Config{}, nil, nil, repository)
+	handler := New(nil, service, nil)
+	w := httptest.NewRecorder()
+	c, _ := gin.CreateTestContext(w)
+	c.AddParam("id", "1")
+	user := &models.User{
+		Groups: []*models.Group{
+			{Name: "name"},
+		},
+	}
+	c.Set("user", user)
 
-	groups := []*models.Group{
-		{
-			Name: name,
+	handler.FindById(c)
+
+	assert.Empty(t, c.Errors)
+	var actualBody model.Database
+	assertResponse(t, w, http.StatusOK, &actualBody, database)
+	repository.AssertExpectations(t)
+}
+
+func TestHandler_Copy(t *testing.T) {
+	userClient := &mockUserClient{}
+	userClient.
+		On("FindGroupByName", "token", "group-name").
+		Return(&models.Group{
+			Name: "group-name",
+		}, nil)
+	s3Client := &mockS3Client{}
+	s3Client.
+		On("Copy", mock.AnythingOfType("string"), "path", "group-name/database-name").
+		Return(nil)
+	repository := &mockRepository{}
+	repository.
+		On("FindById", uint(1)).
+		Return(&model.Database{
+			Url: "/path",
+		}, nil)
+	repository.
+		On("Create", mock.AnythingOfType("*model.Database")).
+		Return(nil)
+	service := NewService(config.Config{}, nil, s3Client, repository)
+	handler := New(userClient, service, nil)
+	w := httptest.NewRecorder()
+	c, _ := gin.CreateTestContext(w)
+	c.AddParam("id", "1")
+	user := &models.User{
+		ID: 1,
+		Groups: []*models.Group{
+			{
+				Name: "group-name",
+			},
 		},
 	}
+	c.Set("user", user)
+	copyRequest := &CopyDatabaseRequest{
+		Name:  "database-name",
+		Group: "group-name",
+	}
+	c.Request = newPost(t, "/groups", copyRequest)
+
+	handler.Copy(c)
+
+	assert.Equal(t, http.StatusCreated, w.Code)
+	assert.Empty(t, c.Errors)
+	userClient.AssertExpectations(t)
+	s3Client.AssertExpectations(t)
+	repository.AssertExpectations(t)
+}
+
+func newPost(t *testing.T, path string, request any) *http.Request {
```

Comment:
```suggestion
func newPost(t *testing.T, path string, jsonBody any) *http.Request {
```

`request` is not accurate in here as it does not represent the HTTP request. `jsonBody` describes it better I think 😄